fidget/src/core/shape/mod.rs

```diff
@@ -213,6 +213,53 @@ impl<F> Shape<F> {
     }
 }
 
+/// Variables bound to values for shape evaluation
+///
+/// Note that this cannot store `X`, `Y`, `Z` variables (which are passed in as
+/// first-class arguments); it only stores [`Var::V`] values (identified by
+/// their inner [`VarIndex`]).
+pub struct ShapeVars<F>(HashMap<VarIndex, F>);
```

Comment:
Is this using a `HashMap` an intermediate step along the way to using a `Vec` / `smallvec`?

Reply:
Perhaps!  I'm inclined to leave it until someone starts noticing it gating performance, which I suspect is a long ways off.